### PR TITLE
[#2680] fix(lakehouse-iceberg): Set a correct nullable value when convert to Iceberg Type from Gravitino Type

### PR DESCRIPTION
**Pull request by caican00**
### What changes were proposed in this pull request?

Set a correct nullable value when convert to Iceberg Type from Gravitino Type.

### Why are the changes needed?
It is a bug.

Fix: https://github.com/datastrato/gravitino/issues/2680

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?
Existing and new unit test.

### CODE REVIEW

**Comment by caican00** [comment]
Hi @jerryshao, could you help review this PR when you are free? Thanks.

**Comment by FANNG1** [comment]
Overall LGTM,  @caican00 could you add a specific unit test in `TestConvertUtil`?

**Comment by caican00** [comment]
> Overall LGTM, @caican00 could you add a specific unit test in `TestConvertUtil`?

image

I fixed the single test in testConvertUtil. Should I add a new unit test？

File: catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/TestConvertUtil.java [hunk 1]
```diff
@@ -161,21 +163,108 @@ public void testToPrimitiveType() {
...
         ConvertUtil.toIcebergType(
-            true,
             com.datastrato.gravitino.rel.types.Types.ListType.of(
                 com.datastrato.gravitino.rel.types.Types.FloatType.get(), true));
     Assertions.assertTrue(listType instanceof Types.ListType);
     Assertions.assertTrue(listType.asListType().elementType() instanceof Types.FloatType);
   }
 
+  @Test
+  public void testToNestedType() {
```
Comment (caican00) [inline catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/TestConvertUtil.java:181]: Added a new unit test to check the complex type

**Comment by caican00** [comment]
> Overall LGTM, @caican00 could you add a specific unit test in `TestConvertUtil`?

added a new unit test in `TestConvertUtil`

File: catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/TestConvertUtil.java [hunk 2]
```diff
@@ -161,21 +163,108 @@ public void testToPrimitiveType() {
...
+                        false)),
+                com.datastrato.gravitino.rel.types.Types.StructType.Field.notNullField(
+                    "col2",
+                    com.datastrato.gravitino.rel.types.Types.ListType.of(
+                        com.datastrato.gravitino.rel.types.Types.FloatType.get(), false))));
+    Assertions.assertTrue(structTypeNotNull instanceof Types.StructType);
+    Assertions.assertTrue(
+        structTypeNotNull.asStructType().fields().get(0).type() instanceof Types.MapType);
+    Assertions.assertTrue(
+        structTypeNotNull.asStructType().fields().get(0).type().asMapType().isValueRequired());
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/TestConvertUtil.java:261]: for struct type, we should check struct filed is nullable not struct field type

**Comment by mchades** [comment]
Overall LGTM

File: catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/TestConvertUtil.java (hunk 2 above)
Comment (caican00) [reply catalogs/catalog-lakehouse-iceberg/src/test/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/TestConvertUtil.java:263]: added a check:
```
    structTypeNullable.asStructType().fields().forEach(f -> Assertions.assertTrue(f.isOptional()));

    structTypeNullable.asStructType().fields().forEach(f -> Assertions.assertTrue(f.isRequired()));
```

**Review by FANNG1 — APPROVED** [review]

**Comment by FANNG1** [comment]
merged to main, @caican00 thanks your work!

**Pull request merged**